heron/tools/cli/src/python/response.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2016 Twitter. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''response.py'''
+import abc
+
+from heron.common.src.python.utils.log import Log
+
+# # pylint: disable=no-init
+class Status(object):
+  """Status code enum"""
+  Ok, NonUserError, UserError = range(3)
+
+class Response(object):
+  """Response class that captures result of executing an action"""
+  def __init__(self, status, succ_msg=None, err_msg=None, extra_msg=None):
+    self.status = self.status_type(status)
+    self.succ_msg = succ_msg
+    self.err_msg = err_msg
+    self.extra_msg = extra_msg
+
+  # Meaning of exit status code:
+  #  - status code = 0:
+  #    program exits without error
+  #  - 0 < status code < 100:
+  #    program fails to execute before program execution. For example,
+  #    JVM cannot find or load main class
+  #  - status code >= 100:
+  #    program fails to launch after program execution. For example,
+  #    topology definition file fails to be loaded
+  @staticmethod
+  def status_type(status_code):
+    if status_code == 0:
+      return Status.Ok
+    elif status_code < 100:
+      return Status.NonUserError
+    else:
+      return Status.UserError
```

Comment:
Yes, I am not satisfied with these two names. `HeronError` and `InvokationError` are fine (note that we have Python topology so `JavaMainError` is not a good name...)